src/app/component/user/components/habit/all-habits/all-habits.component.ts

```diff
@@ -8,30 +11,88 @@ import { Component, OnInit, OnDestroy } from '@angular/core';
   styleUrls: ['./all-habits.component.scss']
 })
 export class AllHabitsComponent implements OnInit, OnDestroy {
-
-  public habitsMockList: object[];
+  private masterSubscription: Subscription = new Subscription();
+  private habitsList: ServerHabitItemPageModel[] = [];
+  public filteredHabitsList: ServerHabitItemPageModel[] = [];
+  public totalHabits: number;
+  public totalHabitsCopy = 0;
   public galleryView = true;
-  private subscription: Subscription;
+  private lang: string;
   public windowSize: number;
+  private batchSize = 6;
+  private currentPage = 0;
+  private totalPages: number;
+  public isFetching = true;
+  public elementsLeft = true;
+  public tagList: string[] = [];
```

Comment:
In the middle of the class, the elements should be arranged in the following order:

- public variables;
- private variables;
- constructor;
- hook methods;
- public methods;
- private methods;
[Style Guide](https://github.com/ita-social-projects/GreenCityClient/blob/docs/ReadMe-style-guide.md)